definition/definition.go

```diff
@@ -182,6 +182,102 @@ func (ac *ActivityConfig) OutputMapper() mapper.Mapper {
 }
 
 type loopCfg struct {
+	Condition  string `md:"condition"`
+	IterateOn  string `md:"iterateOn"`
+	Delay      int    `md:"delay"`
+	Accumulate bool   `md:"accumulate"`
+}
+
+type loop struct {
+	condition  expression.Expr
+	accumulate bool
+	delay      int
+	iterateOn  interface{}
+}
+
+func NewLoop(cfg *loopCfg, ef expression.Factory) (*loop, error) {
+	loop := &loop{}
+	var err error
+	loop.accumulate = cfg.Accumulate
+	loop.delay = cfg.Delay
+
+	if cfg.Condition != "" {
+		if cfg.Condition[0] == '=' {
+			loop.condition, err = ef.NewExpr(cfg.Condition[1:])
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if cfg.IterateOn != "" {
+		if cfg.IterateOn[0] == '=' {
+			loop.iterateOn, err = ef.NewExpr(cfg.IterateOn[1:])
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			loop.iterateOn = cfg.IterateOn
+		}
+	}
+
+	return loop, nil
+}
+
+func (l *loop) Accumulated() bool {
+	return l.accumulate
+}
+
+func (l *loop) DowhileCondition() expression.Expr {
+	return l.condition
+}
+
+func (l *loop) GetIterate() interface{} {
+	return l.iterateOn
+}
+
+func (l *loop) IterateEnabled() bool {
+	return l.iterateOn != nil
+}
+
+func (l *loop) DowhileEnabled() bool {
+	return l.condition != nil
+}
+
+func (l *loop) Delay() int {
+	return l.delay
+}
+
+//Temporary Methods. Need to Remove
+func (l *loop) RetryOnErrorEnabled() bool {
+	return false
+}
+
+//Temporary Methods. Need to Remove
+func (l *loop) RetryOnErrorCount() int {
+	return 0
+}
+
+//Temporary Methods. Need to Remove
+func (l *loop) RetryOnErrorInterval() int {
```

Comment:
Retry on error is independent of the loop, so we should consider to remove it from the loop

Reply:
Yes. This is just for backward compatibility and serves no purpose currently.

Reply:
I would remove those methods.. no body directly calls those outside of flow itself.